### DISCUSSION TITLE
To what extent does the hover tooltip support doxygen style comments?

### DISCUSSION THREAD

**Discussion by Kyr27** [microsoft/vscode-cpptools#12856] [Q&A] [answered]
Hello, i've been trying to get hover info to show an example usage of a function, however it seems like either the functionality is missing or its very much broken.
```
/**
 * @brief Adds two integers together.
 * 
 * @param a The first integer.
 * @param b The second integer.
 * @return int The sum of `a` and `b`.
 * 
 * Example (broken formatting and syntax support)
 * ```cpp
 * int result = add(2, 3);
 * std::cout << result << std::endl; // 5
 * ```
 */
```
```
int add(int a, int b) {
    return a + b;
}
```

The example section breaks formatting, and the resulting code is all mangled into one or multiple lines depending on how large it is without any syntax color support, here's what the hover information shows when you hover over the function in vs code:
int add(int a, int b)
Adds two integers together.

**Parameters:**
```a``` – The first integer.
```b``` – The second integer.

**Returns:**
int The sum of ```a``` and ```b```.

Example ```cpp int result = add(2, 3); std::cout << result << std::endl; // 5```

This type of comment also does not work, in this case, the cpp code block is ignored completely (only the adds 2 together is outputted):
```
/**
 * @brief adds 2 together
 * 
 * ~~~~~{.cpp}
 * add(2, 2);
 * ~~~~~
 */
```

This also results in improper formatting and missing syntax support:
```
/**
 * @brief adds 2 numbers togehter
 * 
 * This function returns the result of adding param a, and param b together
 * 
 * Example usage:
 * @code
 *  int res = add (2, 2);
 *  std::cout << res << '\n';
 * @endcode
 */
```

The only one that works is this one, but if i try to add things with ```@```, such as ```@brief``` or ```@param```, it too breaks formatting and syntax support:
Working version:
```
/// adds 2 numbers together
/// Working Example:
/// ```cpp
/// int a = 5;
/// int b = 10;
/// add(a, b);
/// ```
```

Broken version:
```
/// @brief adds 2 numbers together
///
/// Broken Example:
/// ```cpp
/// int a = 5;
/// int b = 10;
/// add(a, b);
/// ```
```
```
int add(int a, int b) {
    return a + b;
}
```

So are the doxygen comment blocks not supported, or is it a bug, i've seen some libraries dodge around this issue by using triple slashes, but they are unable to use ```@``` as a consequence, which is unacceptable.

**Comment by bobbrow** [comment, marked as answer]
Hi @Kyr27, what you're reporting in your first example is a bug and was previously reported at #12794. We are working on this right now. It's a regression from a previous release. Your 2nd and 3rd examples are not currently supported, but if you like, we can keep this issue open to track a request for support for those formats.

**Reply by Kyr27** [reply]
Thank you for your reply, this is the answer i was looking for, the support for the formats from my 2nd and 3rd examples is not really an issue for me, as long as at least one of the formats is supported, because my code comments look like a bit of a mess currently.
